packages/js/create-product-editor-block/index.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+const { join } = require( 'path' );
+
+module.exports = {
+	defaultValues: {
+		namespace: 'extension',
+		slug: 'example-product-editor-block',
+		version: '0.1.0',
+		category: 'widgets',
+		title: 'Example Product Editor Block',
+		description: 'A block to demonstrate extending the Product Editor',
+		dashicon: 'flag',
+		license: 'GPL-3.0+',
+		attributes: {
+			message: {
+				type: 'string',
+				source: 'text',
+				selector: 'div',
+			},
+		},
+		supports: {
+			html: false,
+			inserter: false,
+		},
+		npmDevDependencies: [
+			'@types/wordpress__block-editor@^7.0.0',
+			'@types/wordpress__blocks@^11.0.9',
+			'@woocommerce/dependency-extraction-webpack-plugin',
+			'@woocommerce/eslint-plugin',
+			'@wordpress/prettier-config',
+			'@wordpress/stylelint-config',
+			'eslint-import-resolver-typescript',
+		],
+		customScripts: {
+			postinstall: 'composer install',
+		},
+		wpScripts: true,
+		wpEnv: true,
+		editorScript: 'file:./index.js',
+		editorStyle: 'file:./index.css',
+		style: 'file:./index.css',
+		customBlockJSON: {
+			script: 'file:./index.js',
```

Comment:
I need to include the `script` prop in my `block.json` in order for the block to load in the product editor

Reply:
Similar to what we discussed in Slack with the latest WP release, I think this is a bug in our initialization code, where only blocks with the `script` are being enqueued.
We can leave this in for now, but I will create a follow up issue/bug to address this.

Reply:
Issue to address this in product editor initialization: https://github.com/woocommerce/woocommerce/issues/38394